From f8458d94e540c610085ab2d3665060248afae268 Mon Sep 17 00:00:00 2001
From: Felipe Eduardo Sanchez Diaz Duran <izto@asic-linux.com.mx>
Date: Wed, 10 Feb 2010 14:44:14 -0600
Subject: [PATCH 01/20] Felipe Sateler:  Tag only regular files as conffiles

Jirka Hladky:    Support for changes in scandir() and scandir64() in glibc-2.10
---
 INSTALL                        |  7 +++----
 checkinstall                   | 10 +++++-----
 description-pak                |  2 +-
 installwatch/create-localdecls | 16 ++++++++++++++++
 installwatch/description-pak   |  1 -
 installwatch/installwatch.c    | 26 ++++++++++++++++++++++++++
 6 files changed, 51 insertions(+), 11 deletions(-)
 delete mode 100644 installwatch/description-pak

diff --git a/INSTALL b/INSTALL
index 40be5c5..2abdf7a 100644
--- a/INSTALL
+++ b/INSTALL
@@ -12,15 +12,14 @@ checkinstall
 "make install" goes first because we have no checkinstall installed yet :)
 
 "checkinstall" goes latter so we can have a properly installed (and removable)
-               package for checkinstall-1.6.1
+               package for checkinstall-1.6.2
 
                (Be sure to read the README if you're planning on building RPM or
                 Debian binary packages)
                 
                
-This will install checkinstall-1.6.1, makepak (modified makepkg) and 
-installwatch-0.7.0beta5
+This will install checkinstall-1.6.2, makepak (modified makepkg) and 
+installwatch-0.7.0
 
 
 ==========
-$Id: INSTALL,v 1.6.0.3 2004/12/01 21:36:17 izto Exp $
diff --git a/checkinstall b/checkinstall
index 41a7978..076aeaa 100755
--- a/checkinstall
+++ b/checkinstall
@@ -1,10 +1,10 @@
 #!/bin/bash
 ##############################################################################
-#         $Id: checkinstall,v 1.6.2.17 2009/12/26 13:15:21 izto Exp $ 
+#         $Id: checkinstall,v 1.6.3.1 2010/02/10 14:42:32 izto Exp $ 
 #                           ########################                         
 #                                                                            
 #
-#                             CheckInstall v1.6.2
+#                             CheckInstall v1.6.3
 #
 #  Installs a compiled program from the program's source directory using     
 #  "make install" or any other command supplied on checkinstall's command  
@@ -38,7 +38,7 @@
 # Trap the INT signal (ctrl-c, for example)
 trap trapint 2
 
-CHECKINSTALL_VERSION=1.6.2
+CHECKINSTALL_VERSION=1.6.3
 
 # locate the file or directory that the parameter corresponds to;
 # always prints absolute path
@@ -2463,7 +2463,7 @@ cd "$DIRECTORIO_FUENTE"
 
 echo
 echogn "Building RPM package..."
-$RPMBUILD -bb ${RPM_TARGET_FLAG}${ARCHITECTURE} "$SPEC_PATH" &> ${TMP_DIR}/rpmbuild.log
+$RPMBUILD -bb ${RPM_TARGET_FLAG}${ARCHITECTURE} "$SPEC_PATH" --buildroot="$BROOTPATH" &> ${TMP_DIR}/rpmbuild.log
 okfail
 
 if [ $? -gt 0 ]; then
@@ -2608,7 +2608,7 @@ if [ -f postremove-pak ]; then
 fi
 
 # Tag files in /etc to be conffiles
-find $BUILD_DIR/etc 2> /dev/null | sed -e "s,$BUILD_DIR,," | \
+find $BUILD_DIR/etc -type f 2> /dev/null | sed -e "s,$BUILD_DIR,," | \
 	grep -v '^/etc$' > $BUILD_DIR/DEBIAN/conffiles
 
 # The package will be saved here (ignores <epoch>: prefix):
diff --git a/description-pak b/description-pak
index d9ace66..01fd583 100644
--- a/description-pak
+++ b/description-pak
@@ -1,4 +1,4 @@
-CheckInstall installations tracker, version 1.6.2a1 - pre git
+CheckInstall installations tracker, version 1.6.2
 
 CheckInstall  keeps  track of all the files created  or
 modified  by your installation  script  ("make install"
diff --git a/installwatch/create-localdecls b/installwatch/create-localdecls
index cce5bc9..f7adf4d 100755
--- a/installwatch/create-localdecls
+++ b/installwatch/create-localdecls
@@ -64,6 +64,7 @@ if test "$VERSION" = 'libc.so.6' ; then
 	case "$OsLibcMajor" in
 	2)
 		# 2 is the glibc version
+		echo "JH OsLibcMinor is $OsLibcMinor"
 		case "$OsLibcMinor" in
 		0)
 			echo '#define GLIBC_MINOR 0' >> $OUTFILE
@@ -89,6 +90,21 @@ if test "$VERSION" = 'libc.so.6' ; then
 		7)
 			echo '#define GLIBC_MINOR 7' >> $OUTFILE
 			SUBVERSION='glibc-2.7' ;;
+		8)
+			echo '#define GLIBC_MINOR 8' >> $OUTFILE
+			SUBVERSION='glibc-2.8' ;;
+		9)
+			echo '#define GLIBC_MINOR 9' >> $OUTFILE
+			SUBVERSION='glibc-2.9' ;;
+		10)
+			echo '#define GLIBC_MINOR 10' >> $OUTFILE
+			SUBVERSION='glibc-2.10' ;;
+		11)
+			echo '#define GLIBC_MINOR 11' >> $OUTFILE
+			SUBVERSION='glibc-2.11' ;;
+		12)
+			echo '#define GLIBC_MINOR 12' >> $OUTFILE
+			SUBVERSION='glibc-2.12' ;;
 		*)
 			echo 'Treated as glibc >= 2.1 (finger crossed)'
 			echo '#define GLIBC_MINOR 1' >> $OUTFILE
diff --git a/installwatch/description-pak b/installwatch/description-pak
deleted file mode 100644
index e8014b9..0000000
--- a/installwatch/description-pak
+++ /dev/null
@@ -1 +0,0 @@
-Package created with checkinstall 1.6.0beta1
diff --git a/installwatch/installwatch.c b/installwatch/installwatch.c
index 21c91c7..8004b5b 100644
--- a/installwatch/installwatch.c
+++ b/installwatch/installwatch.c
@@ -98,9 +98,20 @@ static int (*true_rename)(const char *, const char *);
 static int (*true_rmdir)(const char *);
 static int (*true_xstat)(int,const char *,struct stat *);
 static int (*true_lxstat)(int,const char *,struct stat *);
+
+#if(GLIBC_MINOR >= 10)
+
+static int (*true_scandir)(	const char *,struct dirent ***,
+				int (*)(const struct dirent *),
+				int (*)(const struct dirent **,const struct dirent **));
+
+#else
+
 static int (*true_scandir)(	const char *,struct dirent ***,
 				int (*)(const struct dirent *),
 				int (*)(const void *,const void *));
+#endif
+
 static int (*true_symlink)(const char *, const char *);
 static int (*true_truncate)(const char *, TRUNCATE_T);
 static int (*true_unlink)(const char *);
@@ -118,9 +129,16 @@ static FILE *(*true_fopen64)(const char *,const char *);
 static int (*true_ftruncate64)(int, __off64_t);
 static int (*true_open64)(const char *, int, ...);
 static struct dirent64 *(*true_readdir64)(DIR *dir);
+
+#if(GLIBC_MINOR >= 10)
+static int (*true_scandir64)(	const char *,struct dirent64 ***,
+				int (*)(const struct dirent64 *),
+				int (*)(const struct dirent64 **,const struct dirent64 **));
+#else
 static int (*true_scandir64)(	const char *,struct dirent64 ***,
 				int (*)(const struct dirent64 *),
 				int (*)(const void *,const void *));
+#endif
 static int (*true_xstat64)(int,const char *, struct stat64 *);
 static int (*true_lxstat64)(int,const char *, struct stat64 *);
 static int (*true_truncate64)(const char *, __off64_t);
@@ -3079,7 +3097,11 @@ int rmdir(const char *pathname) {
 
 int scandir(	const char *dir,struct dirent ***namelist,
 		int (*select)(const struct dirent *),
+#if (GLIBC_MINOR >= 10)
+		int (*compar)(const struct dirent **,const struct dirent **)	) {
+#else
 		int (*compar)(const void *,const void *)	) {
+#endif
 	int result;
 
 	if (!libc_handle)
@@ -3691,7 +3713,11 @@ struct dirent64 *readdir64(DIR *dir) {
 
 int scandir64(	const char *dir,struct dirent64 ***namelist,
 		int (*select)(const struct dirent64 *),
+#if (GLIBC_MINOR >= 10)
+		int (*compar)(const struct dirent64 **,const struct dirent64 **)	) {
+#else
 		int (*compar)(const void *,const void *)	) {
+#endif
 	int result;
 
 	if (!libc_handle)
-- 
2.30.0

